From 6f903f6870011b6a11074d93a03481faac6a35dc Mon Sep 17 00:00:00 2001
From: dseefeld <dseefeld@microsoft.com>
Date: Thu, 4 Feb 2021 15:18:11 +0000
Subject: [PATCH] Exclude some projects from source-build

---
 Directory.Build.props | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Directory.Build.props b/Directory.Build.props
index 50d30f12fd..85c2f2428b 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -27,7 +27,9 @@
         '$(IsTestAssetProject)' == 'true' OR
         '$(IsBenchmarkProject)' == 'true' OR
         '$(IsUnitTestProject)' == 'true'">false</IsShipping>
-
+    <IsOtherTestProject Condition="$(MSBuildProjectName.Contains('IntegrationTesting')) or $(MSBuildProjectName.Contains('TestCommon'))">true</IsOtherTestProject>
+    <ExcludeFromSourceBuild Condition="'$(IsSampleProject)' == 'true' or '$(IsTestAssetProject)' == 'true' or '$(IsBenchmarkProject)' == 'true' or '$(IsUnitTestProject)' == 'true' or '$(IsSpecificationTestProject)' == 'true' or '$(IsOtherTestProject)' == 'true'">true</ExcludeFromSourceBuild>
+
     <!--
       Following logic mimics core-setup approach as well as
       https://github.com/dotnet/arcade/blob/694d59f090b743f894779d04a7ffe11cbaf352e7/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj#L30-L31
-- 
2.21.3

