From d67aa83be72cbb1ca37aba992633da87235f397b Mon Sep 17 00:00:00 2001
From: Tomas Weinfurt <tweinfurt@yahoo.com>
Date: Wed, 13 Nov 2019 22:32:02 -0800
Subject: [PATCH 06/14] Add FreeBSD

---
 Directory.Build.props | 1 +
 eng/Common.props      | 1 +
 2 files changed, 2 insertions(+)

diff --git a/Directory.Build.props b/Directory.Build.props
index cd2dabcdba..53d90f2903 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -159,6 +159,7 @@
       win-arm64;
       osx-x64;
       osx-arm64;
+      freebsd-x64;
       linux-musl-x64;
       linux-musl-arm;
       linux-musl-arm64;
diff --git a/eng/Common.props b/eng/Common.props
index 7c489b15a9..a9a69bde9f 100644
--- a/eng/Common.props
+++ b/eng/Common.props
@@ -3,6 +3,7 @@
     <TargetOsName Condition=" '$(TargetOsName)' == '' AND $([MSBuild]::IsOSPlatform('Windows'))">win</TargetOsName>
     <TargetOsName Condition=" '$(TargetOsName)' == '' AND $([MSBuild]::IsOSPlatform('OSX'))">osx</TargetOsName>
     <TargetOsName Condition=" '$(TargetOsName)' == '' AND $([MSBuild]::IsOSPlatform('Linux'))">linux</TargetOsName>
+    <TargetOsName Condition=" '$(TargetOsName)' == '' AND $([MSBuild]::IsOSPlatform('FreeBSD'))">freebsd</TargetOsName>
     <TargetArchitecture Condition="'$(TargetArchitecture)' == ''">x64</TargetArchitecture>
     <TargetRuntimeIdentifier Condition="'$(TargetRuntimeIdentifier)' == ''">$(TargetOsName)-$(TargetArchitecture)</TargetRuntimeIdentifier>
   </PropertyGroup>
-- 
2.18.0

