From 446691a345e68861bd653c1be4b9111aafad5613 Mon Sep 17 00:00:00 2001
From: dseefeld <dseefeld@microsoft.com>
Date: Thu, 8 Oct 2020 14:18:02 +0000
Subject: [PATCH 12/14] Don't warn on CA1416

---
 Directory.Build.props | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Directory.Build.props b/Directory.Build.props
index 53d90f2903..72cee68b4e 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -102,6 +102,9 @@
 
   <!-- Warnings and errors -->
   <PropertyGroup>
+    <!-- Don't warn on CA1416 since source-build TFM changes
+        are causing the warning to occur -->
+    <NoWarn>$(NoWarn);CA1416</NoWarn>
     <!-- Ensure API docs are available. -->
     <NoWarn>$(NoWarn.Replace('1591', ''))</NoWarn>
 
-- 
2.18.0

